File: CSharp/AnalyseRecord.cs
using CanEasy;

namespace MyApplication
{
    public class Application : BaseApplication
    {
        protected override void OnStarting()
        {
			// Create filter to process message record entries
            var filter = CanEasy.Record.CreateFilter(RecordEntryType.MsgRecordEntry);
			// Create iterator to and walk over the record
            var iter = CanEasy.Record.CreateIterator(filter);
            while(iter.Next())
            {
				// Write information about the record entry
                var entry = (MsgRecordEntry)iter.RecordEntry;
                var sData = System.BitConverter.ToString((byte[])entry.Data);
                CanEasy.MakeReport($"TS={(entry.Timestamp*3600).ToString("000.0000")}, Bus={entry.Bus.Name}, Id=0x{entry.Id.Plain.ToString("x")}, Rx={entry.Rx}, Data={sData}", ReportType.ReportTypeInformation);
            }

			// Stop application
            Exit();
        }
    }
}

File: Python/AnalyseRecord.py
from caneasylib import *
import win32com.client as win32com

@entry
class Application(BaseApplication):
    def OnStarting(self):
        # Get CanEasyApplication instance
        caneasy:ICanEasyApplication = getCanEasy()
        
        # Create filter to process message record entries
        filter = caneasy.Record.CreateFilter(constants.MsgRecordEntry) 
        # Create iterator to and walk over the record
        iter = caneasy.Record.CreateIterator(filter)
        while iter.Next():
            # Write information about the record entry
            entry = win32com.CastTo(iter.RecordEntry, "IMsgRecordEntry")
            caneasy.MakeReport("TS={:-010.4f}, Bus={}, Id=0x{:x}, Rx={}, Data={}".format(entry.Timestamp*3600, entry.Bus.Name, entry.Id.Plain, entry.Rx, bytes(entry.Data).hex()), constants.ReportTypeInformation)

        # Stop application
        exit()

